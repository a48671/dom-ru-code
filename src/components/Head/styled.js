import styled from 'styled-components';
import { colors } from '../../variables/colors';

export const Wrapper = styled.div.attrs({ className: 'head' })`
  display: flex;
  align-items: center;
  width: 100%;
  padding: 10px 20px;
`;

export const Logo = styled.div.attrs({ className: 'head__logo' })`
  width: 70px;
  height: 70px;
  img {
    max-width: 100%;
    max-height: 100%;
  }
`;

export const Title = styled.h1.attrs({ className: 'head__title' })`
  width: calc(100% - 70px);
  font-size: 24px;
  font-weight: 600;
  padding-left: 30px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  a {
    color: ${colors.blue};
    text-decoration: underline;
  }
  @media (max-width: 430px) {
    font-size: 22px;
  }
`;
