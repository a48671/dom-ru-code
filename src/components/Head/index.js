import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Wrapper, Logo, Title } from './styled';

class Head extends PureComponent {
  render() {
    const { title, logo, url } = this.props;

    return (
      <Wrapper>
        <Logo>
          <img src={logo} alt={title} />
        </Logo>
        <Title>
          <a href={url} target="_blank" rel="noopener noreferrer">
            {title}
          </a>
        </Title>
      </Wrapper>
    );
  }
}

Head.propTypes = {
  title: PropTypes.string,
  logo: PropTypes.string,
  url: PropTypes.string,
};

Head.defaultProps = {
  title: '',
  logo: '',
  url: '',
};

export default Head;
