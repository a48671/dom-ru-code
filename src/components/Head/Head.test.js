import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Head from './index';

describe('Head', () => {
  it('render correctly component', () => {
    const HeadComponent = ReactTestRenderer.create(
      <Head title="Title" logo="#" url="#" />
    ).toJSON();
    expect(HeadComponent).toMatchSnapshot();
  });
});
