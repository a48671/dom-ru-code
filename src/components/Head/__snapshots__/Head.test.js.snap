// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head render correctly component 1`] = `
<div
  className="sc-bdVaJa head dCvSV"
>
  <div
    className="sc-bwzfXH head__logo JJPRh"
  >
    <img
      alt="Title"
      src="#"
    />
  </div>
  <h1
    className="sc-htpNat head__title jTvJLX"
  >
    <a
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      Title
    </a>
  </h1>
</div>
`;
